engine/games/picross_minigame.asm

```diff
@@ -450,34 +489,37 @@ PicrossMinigame:
 	ret
 
 .HandleError:
+; If something's gone wrong, print a big ol' "ERROR" in
+; the middle of the game display.
 	ld a, [wPicrossErrorCheck]
 	and a
 	ret z
 
-; If something's gone wrong, print a big ol' "ERROR" in
-; the middle of the game display.
 	ld hl, .ErrorBitmap
 	decoord 0, 9
 	ld c, $64
 
 .print_message:
-	ldi a, [hl]
+	ld a, [hli]
 	add a, $80
 	ld [de], a
 	inc de
 	dec c
 	jr nz, .print_message
 	ret
-; End of function .HandleError
 
 .ErrorBitmap:
-	db 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0
-	db 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0
-	db 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0
-	db 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0
-	db 0, 1, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0
-
-PicrossMinigame_CheckPuzzleSolved:
+	pushc
+	charmap " ", 0
+	charmap "█", 1
+	db " ██ ██  ██  ███ ██  "
+	db " █  █ █ █ █ █ █ █ █ "
+	db " ██ ██  ██  █ █ ██  "
+	db " █  █ █ █ █ █ █ █ █ "
+	db " ██ █ █ █ █ ███ █ █ "
+	popc
```

Comment:
Not very sure if this is the right way to go about it
This bit renders out the ERROR text as `db 0`'s and `db 1`'s

Reply:
That's a good solution!